src/qibocal/auto/task.py

```diff
@@ -100,13 +106,19 @@ def run(self, folder: Path, platform: Platform, qubits: Qubits) -> Results:
             parameters = DummyPars()
 
         path = self.datapath(folder)
-
         if operation.platform_dependent and operation.qubits_dependent:
-            if self.qubits:
-                qubits = allocate_qubits(platform, self.qubits)
-            self._data: Data = operation.acquisition(
-                parameters, platform=platform, qubits=qubits
-            )
+            if platform is not None:
+                if self.action.qubits:
+                    qubits = allocate_qubits(platform, self.action.qubits)
+            if qubits:
+                self._data: Data = operation.acquisition(
+                    parameters, platform=platform, qubits=qubits
+                )
+                # after acquisition we update the qubit parameter
+                self.qubits = list(qubits.keys())
```

Comment:
```suggestion
                self.qubits = list(qubits)
```
`list()` is creating a list out of an iterable. The default iteration on `dict`s return keys :)